torch/csrc/autograd/functions/utils.h

```diff
@@ -42,9 +42,6 @@ struct ComputeRequiresGrad : IterArgs<ComputeRequiresGrad> {
 
 template <typename... Args>
 inline bool compute_requires_grad(Args&&... args) {
-  if (!GradMode::is_enabled()) {
```

Comment:
Why is this removed?
It looks like this function is used in most auto-generated backward, is there a performance impact on adding more indirection to it?

Reply:
The idea is to move `GradMode` check to `Varaible::requires_grad`, so we can use it in ATen. Good point on the performance, but given that our functions usually take 1~4 inputs, I think extra 2~3 bool checks per function is not a big problem :)

Reply:
Huh? So variable.requires_grad() will return false if GradMode isn't enabled even if the variable requires grad? That doesn't seem like a good idea

Reply:
The short-circuit works the other way. If GradMode is off, then it'll have to check *every* input as currently written in this PR.

But the more important issue is that this changes the semantics of requires_grad(), which I don't think we should be doing.

Reply:
@colesbury Yes, that is indeed the part I am not so sure about, but it would enable optimizations that would be otherwise impossible because `GradMode` lives in `torch::`.

Reply:
Can't you move GradMode out of torch without changing the semantics of requires_grad?